### DISCUSSION TITLE
Release bitcoin v0.31.0

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1981] [General]
Release of bitcoin v0.31.0 is due at the end of the month.  This is a list of potential PRs to merge to get there.

- Release `bech32 v0.10.0-beta` - DONE
- Release `hashes v0.13.0` - DONE
- Release `secp256k1 v0.28.0` - DONE

### Release bitcoin v0.31.0-rc1

- https://github.com/rust-bitcoin/rust-bitcoin/pull/2107
- https://github.com/rust-bitcoin/rust-bitcoin/pull/2118

### Release bitcoin v0.31.0-rc2

- https://github.com/rust-bitcoin/rust-bitcoin/pull/1880
- Add changelog notes created by github actions when we tag rc1 release. 

...

### Release bitcoin v0.31.0

- https://github.com/rust-bitcoin/rust-bitcoin/pull/2099

**Comment by tcharding** [comment]
We are waiting on secp release so I added a few more interesting PRs to the todo list.

**Comment by tcharding** [comment]
I've moved the example PR down to nice-to-haves so it doesn't hold us up.

**Comment by clarkmoody** [comment]
Should we use the GitHub Projects view to create boards for the upcoming releases?

**Reply by tcharding** [reply]
That's a good idea. I've only really come around to liking github in the last year or two, any tips on better ways to use it are most welcome. I'll have a play with it.

**Reply by yancyribbens** [reply]
Is there a different open source tool that we could use instead that would be as good?  Personally over the last few years i've come to not like github so much for a number of reasons.

**Comment by tcharding** [comment]
Closing in favour of https://github.com/orgs/rust-bitcoin/projects/6

**Comment by tcharding** [comment]
The new release project seems to have a "private" label on it, can someone without maintainer privileges please confirm if they can see it or not - I did not intend to do any project management in private.

**Reply by tcharding** [reply]
Oh organisation owners can change the visibility in settings, sorry @apoelstra can you do that at some stage please.

**Reply by tcharding** [reply]
https://github.com/orgs/rust-bitcoin/projects/6/settings

**Comment by tcharding** [comment]
I'm not finding https://github.com/orgs/rust-bitcoin/projects/6/views/1?visibleFields=%5B%22Title%22%2C%22Status%22%2C%22Repository%22%5D useful, perhaps I just can't use the tool but since project management is not the way I choose to spend my day I'm going to go with the tool that is the least painful for me to use. Re-opening this discussion to track what we are doing with the upcoming release. This is informative only, to the best of my understading, @apoelstra is the BDFL, I just do this on because it seems like a good use of the grant money and no one likes doing it.

**Reply by apoelstra** [reply]
>seems like a good use of the grant money and no one likes doing it

Thank you for doing it!

While I may be BDFL, tbh if somebody else comes up with a release plan and pushes on it, probably I'll just go with it because it saves me the chore tasks.

**Reply by tcharding** [reply]
I guessed that was your view :) Happy to carry the load.
